### ISSUE TITLE
Custom node_modules location via modules-folder in package.json

### ISSUE THREAD

**Issue by meridius** [yarnpkg/yarn#2782]
**Do you want to request a *feature* or report a *bug*?**
feature complementing already existing and working `install` option

**What is the current behavior?**
There is no way to set Yarn to permanently (per project) install dependencies to custom directory.  

**What is the expected behavior?**
Although custom install directory can be set on CLI via `yarn install --modules-folder vendor/node_modules` there should be a section in `package.json` file so `--modules-folder` doesn't have to be specified all the time.

I propose to add new section to `package.json` root named `modules-folder`. This would hold string value representing path where dependencies should be installed (same as when specified on CLI) relative to the `package.json` file.

User should be able to override the setting by specifying `--modules-folder` on CLI.  
Also they should be able to use `yarn config` to work with the setting.  
And lastly the <https://yarnpkg.com/en/docs/package-json> shoud be updated.

**Please mention your node.js, yarn and operating system version.**
yarn 0.20.3, Alpine Linux

Thanks for consideration.

**Comment by bestander** [comment]
I think all flags should be configurable via .yarnrc.
Not sure if we should allow yarn specific config via package.json

**Comment by meridius** [comment]
Maybe, but even using `yarn config set modules-folder vendor/node_modules` does not make it use the specified directory. Is it not implemented yet?

Btw, there is only one brief mention of `.yarnrc` file on https://yarnpkg.com

**Comment by bestander** [comment]
Yeah, it is not yet implemented.
Great if you send a PR :)

**Comment by pimvdmolen** [comment]
We could really use this feature!

**Comment by dereuromark** [comment]
I second that!
If https://bower.io/blog/2017/how-to-migrate-away-from-bower/ the please with the same config in the json config file.

**Comment by ryanfb** [comment]
I tried looking into why the config set by `yarn config set modules-folder vendor/node_modules` wasn't being used, and it's because this results in the line `modules-folder vendor/node_modules` in the `.yarnrc`. Since [yarn's RC loading code looks for lines prefixed with `--` to set config options](https://github.com/yarnpkg/yarn/blob/002d39c9b18d9e6e62deb0fe8fc4dc0f3292ccb1/src/rc.js#L38), if you manually edit this line to instead read `--modules-folder vendor/node_modules`, `yarn install` will use it correctly.

If you try `yarn config set --modules-folder vendor/node_modules` you currently get: `error Invalid subcommand. Try "set, get, delete, list, current"`.

**Comment by surminus** [comment]
@ryanfb I just stumbled across this and the correct way to update it using the CLI was:

`yarn config set modules_folder vendor/node_modules`

**Comment by Tigerman55** [comment]
@himdel I think you need to use ./vendor/node_modules

**Comment by paxperscientiam** [comment]
@himdel This seems to do absolutely nothing in version `1.10.1`.

Is it supposed to write to `.yarnrc`?

**Comment by himdel** [comment]
@paxperscientiam Yup :)...

```diff
--- ~/.yarnrc-before	2018-11-02 11:41:22.704147321 +0000
+++ ~/.yarnrc-after	2018-11-02 11:41:11.476046733 +0000
@@ -2,5 +2,6 @@
 # yarn lockfile v1
 
 
+"--modules-folder" vendor/node_modules
 lastUpdateCheck 1541085232301
 save-prefix "~"
```

**Comment by paxperscientiam** [comment]
@himdel Yeah, i figured out some since: https://github.com/yarnpkg/yarn/issues/6594#issuecomment-436387507
